dali/c_api/operator_trace_test.cc

```diff
@@ -139,11 +145,14 @@ TEST_P(OperatorTraceTest, OperatorTraceTest) {
     for (int i = 0; i < prefetch_depth; i++) {
       daliShareOutput(&h);
 
-      for (const auto & operator_name : operator_under_test_names) {
-        EXPECT_EQ(daliHasOperatorTrace(&h, operator_name.c_str(), "this_trace_does_not_exist"), 0);
-        ASSERT_NE(daliHasOperatorTrace(&h, operator_name.c_str(), "test_trace"), 0);
+      for (size_t op = 0; op < std::size(operator_under_test_names); op++) {
+        const char *operator_name = operator_under_test_names[op];
+        const char *trace_name = operator_trace_names[op];
+        EXPECT_EQ(daliHasOperatorTrace(&h, operator_name, "this_trace_does_not_exist"), 0);
+        ASSERT_NE(daliHasOperatorTrace(&h, operator_name, trace_name), 0)
```

Comment:
The test was passing, because the trace name was the same in all operators. Now there are different trace names so we know that we're looking at the right operator's trace.